### PR TITLE
add caret tidiers

### PR DESCRIPTION
**Pull request by mkuehn10**
implements #270

### CODE REVIEW

**Comment by mkuehn10** [comment]
Failure message was
`Packages suggested but not available: 'gamlss' 'ks' 'rgeos'
The suggested packages are required for a complete check.
Checking can be attempted without them by setting the environment
variable _R_CHECK_FORCE_SUGGESTS_ to a false value.
See section 'The DESCRIPTION file' in the 'Writing R Extensions'
manual.`

I don't think my PR has anything to do with those packages,  I added `caret` to the DESCRIPTION for Suggests.

File: R/caret_tidiers.R [hunk 1]
```diff
@@ -0,0 +1,111 @@
+#' Tidying methods for confusionMatrix objects
+#'
+#' Tidies the result of confusion matrix from the \code{caret} package.
+#' Only a \code{tidy} method is provided, not an \code{augment} or
+#' \code{glance} method.
+#'
+#' @param x An object of class \code{confusionMatrix}
+#' @param show_class A boolean of whether to show the values for class specific
```
Comment (alexpghayes) [inline R/caret_tidiers.R]: I'd use "logical" instead of boolean since that's more standard for R.

Comment (alexpghayes) [reply R/caret_tidiers.R]: I'd also rename `show_class` to `by_class`.

File: R/caret_tidiers.R [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+#' quantities from the confusion matrix (specificty, sensitivity, etc.). If set
+#' to FALSE, result will only show accuracy and kappa.
+#' @param ... extra arguments (not used)
+#'
+#' @return A data.frame with one or more of the following columns:
```
Comment (alexpghayes) [inline R/caret_tidiers.R]: Should return a tibble!

Comment (mkuehn10) [reply R/caret_tidiers.R]: I think I need to fix the durbinWatsonTest as well then.

**Comment by mkuehn10** [comment]
It's giving me errors on my test script saying that str_to_lower couldn't be found, but it's from the stringr package.

**Comment by atyre2** [comment]
WRT to builds failing on travis-ci, this message is at the end of all the failures I"ve just checked:
The job exceeded the maximum time limit for jobs, and has been terminated.
It looks like it is taking a very long time to get the machine set up, so never gets around to testing the package.

**Comment by alexpghayes** [comment]
At the moment it's okay for the Travis build to fail, that's on us, but AppVeyor does need to pass. Pretty busy the next couple of days but will get back to this at some point this week.

**Comment by mkuehn10** [comment]
Thanks @atyre2 I saw that as well.  Thanks for the response @alexpghayes, I'll do whatever is needed to fix this, but I'm really not sure what's going on.

**Comment by alexpghayes** [comment]
Awesome! Thanks for your persistence with this!

**Pull request merged**